common/persistence/visibility/store/elasticsearch/processor.go

```diff
@@ -423,16 +423,12 @@ func (a *ackChan) recordStart(metricsClient metrics.Client) {
 }
 
 func (a *ackChan) done(ack bool, metricsClient metrics.Client) {
-	select {
-	case a.ackChInternal <- ack:
-		doneAt := time.Now().UTC()
-		if !a.createdAt.IsZero() {
-			metricsClient.RecordTimer(metrics.ElasticsearchBulkProcessor, metrics.ElasticsearchBulkProcessorRequestLatency, doneAt.Sub(a.createdAt))
-		}
-		if !a.startedAt.IsZero() {
-			metricsClient.RecordTimer(metrics.ElasticsearchBulkProcessor, metrics.ElasticsearchBulkProcessorCommitLatency, doneAt.Sub(a.startedAt))
-		}
-	default:
-		metricsClient.IncCounter(metrics.ElasticsearchBulkProcessor, metrics.ElasticsearchBulkProcessorDeadlock)
+	a.future.Set(ack, nil)
```

Comment:
previous code allow done to be called multiple times. Future does not allow that. Will that be a problem?

Reply:
IMO, the previous code allow but it is also unexpected. It emits a deadlock metrics. I double check the call path. I don't think the done() will be called more than once

Reply:
Yes, I added this metric because I had a suspicious but it never happened and I have never seen any emits for it. 

`ElasticsearchBulkProcessorDeadlock` metric should be removed also.